app/components/goodcity/offers-search-overlay.js

```diff
@@ -50,14 +97,7 @@ export default Ember.Component.extend(SearchMixin, {
         .query("offer", params)
         .then(offers => {
           if (this.get("isMobileApp") && offers.get("length") == 1) {
```

Comment:
The idea that a pagination query method triggers a popup for a **specific** action INSIDE a **reusable** component is a big no-no. That logic belongs to the code that _triggers_ the overlay. (separation of concerns)

It needs to be fixed asap, because as of now this component is unusable anywhere else in the app.

I shouldn't have been like this to begin with. You can check with @abulsayyad123, he knows that code and he might know why this exists. Feel free to drop me a message if there's any questions :)

Reply:
Will discuss with @abulsayyad123 for the same